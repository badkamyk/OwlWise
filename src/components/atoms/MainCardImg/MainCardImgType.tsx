export type MainCardImageType = {
    src: string;
    alt: string;
    className?: string;
}