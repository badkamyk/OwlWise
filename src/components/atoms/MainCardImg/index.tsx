import { MainCardImageType } from './MainCardImgType'

export default function MainCardImg({ src, alt, className }: MainCardImageType) {
    return (
        <img src={src} alt={alt} className={className} />
    )
}
