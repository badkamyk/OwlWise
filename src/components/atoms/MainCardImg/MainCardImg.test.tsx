import { render, screen } from "@testing-library/react";
import { expect, it } from "vitest";
import MainCardImg from "./index";

it("MainCardImg with props", () => {
	const { container } = render(
		<MainCardImg
			src="https://via.placeholder.com/150"
			alt="placeholder"
			className="h-32 w-32"
		/>
	);
	expect(screen.getByAltText("placeholder")).toBeTruthy();
	expect(container.getElementsByClassName("h-32 w-32")).toBeTruthy();
	expect(container).toMatchSnapshot();
});
