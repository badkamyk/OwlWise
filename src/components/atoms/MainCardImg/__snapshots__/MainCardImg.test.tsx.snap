// Vitest Snapshot v1

exports[`MainCardImg with props 1`] = `
<div>
  <img
    alt="placeholder"
    class="h-32 w-32"
    src="https://via.placeholder.com/150"
  />
</div>
`;

exports[`renders MainCardImg with props correctly 1`] = `
<div>
  <img
    alt="placeholder"
    class="h-32 w-32"
    src="https://via.placeholder.com/150"
  />
</div>
`;
